scanner/Makefile

```diff
@@ -19,15 +19,39 @@ else
 QUAY_TAG_EXPIRATION=never
 endif
 
+# CGO is disabled for Scanner V4 builds because:
+#
+# 1. Local builds aren't dockerized. When e.g. host OS's glibc does not match version in target container, scanner
+# refuses to start with errors looking like this:
+#    $ /usr/local/bin/scanner --help
+#    /usr/local/bin/scanner: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /usr/local/bin/scanner)
+#    /usr/local/bin/scanner: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /usr/local/bin/scanner)
+#
+# 2. GitHub builds for arm64, ppc64le and s390x need support for cross-compilation which, when unavailable, makes
+# builds fail with errors like
+#    gcc_arm64.S:30: Error: no such instruction: `stp x29,x30,[sp,'
+#    gcc_arm64.S:34: Error: too many memory references for `mov'
+#
+# This flag must be overridden in commercial/downstream builds in arguments to `make`.
+#    make CGO_ENABLED=1 blah          # Correct - flips it to 1.
+#    CGO_ENABLED=1 make blah          # Wrong! This has no effect. Don't do this.
+#    export CGO_ENABLED=1; make blah  # Wrong! This has no effect. Don't do this.
+# Similarly, don't do the following in Dockerfile
+#    ENV CGO_ENABLED=1                # Wrong!..
+#    RUN make blah                    # because it has no effect. Don't do this.
+# If you need to separate the declaration and the usage in Dockerfile, do this instead
+#    ARG CGO_ENABLED=1
+#    RUN make CGO_ENABLED=$CGO_ENABLED blah
+# TODO: dockerize local builds and flip this to 1.
```

Comment:
We will not do that in the short term. The Scanner team uses local builds to perform UTs and run Scanner with debugging enabled, and likely, if we support dockerized builds, we will continue to make an effort to support non-container builds. Let's remove this TODO?